File: Chapter03/3.3.2 OCP/BadWay/Midfielder.cs
namespace Ocp.BadWay
{
    public class Midfielder
    {
        public void ExecutePlay(string playName)
        {
            if (playName == "DribblePastOpponent")
            {
                Console.WriteLine("Executing a dribble move...");
                // Dribbling logic here
            }
            else if (playName == "DefensiveFormation")
            {
                Console.WriteLine("Getting into defensive position...");
                // Defensive logic here
            }
            // To add a new play, we MUST add another "else if" here.
        }
    }
}

File: Chapter 03/Java/src/main/java/com/grokkingsoftwarearchitecture/chapter03/3.3.2 OCP/Before/MidFielder.java
package com.grokkingsoftwarearchitecture.chapter03.ocp.before;

public class Midfielder {
    public void executePlay(String playName) {
        if ("DribblePastOpponent".equals(playName)) {
            System.out.println("  [Action] Executing a dribble move…");
        } else if ("DefensiveFormation".equals(playName)) {
            System.out.println("  [Action] Getting into defensive position…");
        } else {
            System.out.println("  [Error] Unknown play: " + playName);
        }
    }
}